lips/lip-10.md

```diff
@@ -0,0 +1,190 @@
+* It should also be possible(and probably going to be most common) for VASPs to submit their desired Libra ID domain(s) during the registration.
+* Association may provide an open source application(*Indexer*) that will allow indexing available Libra IDs and will provide a convenient REST api for applications to fetch information about the domains. The API of such an indexer is out of scope of this RFC.
+
+# Peer-2-peer payment protocol
+One of the main use cases for Libra ID is transferring money from one user to another across different wallets.
+
+## UI flow
+* Receiver shares their Libra ID with sender
+* Sender enters receiver’s Libra ID(and other txn details such as amount) in their wallet app
+* Sender’s wallet can fetch profile picture and full name of the receiver prior to sending money, and show it to sender, to improve security (if receiver VASP allows it)
+* Sender presses ‘send’ button
+* Receiver receives money and information about the sender (optionally)
+
+## Backend flow
+In order to support peer-2-peer payments, VASPs need to implement two Libra ID endpoints:
+* **Info endpoint** for querying optional information about the receiver. VASPs can provide this endpoint to improve UX and security of payment, but they are not obligated to - VASPs may choose not to share any information about the user via this endpoint, including whether the user exists or not
+* **Payment endpoint** to perform actual money transfer.
+
+Common considerations:
+
+* For both endpoints, all communications are signed via the same mechanism as for other off-chain communications, see the [Payloads section of lip-1](https://github.com/libra/lip/blob/master/lips/lip-1.mdx#payloads) for details. As such, all vasp-2-vasp communications are authenticated.
+
+* All Libra ID endpoints are nested under `pay.<version>` URI path:
+
+     `https://<hostname>:<port>/pay.<protocol_version>/<endpoint>`
+
+* All Libra ID endpoints use the same `base_url` and same `compliance_public_key` for authentication as other off-chain APIs.
+
+* All Libra ID endpoints are HTTP POST endpoints, they accept JSON-encoded requests and return JSON-encoded responses. Other encoding methods can be considered in the future, `Content-Type` and `Accept` HTTP headers can be used to define request/response formats.
+
+## Info endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/info
+```
+
+**Info endpoint** is used by the **Sender** to request information about the **Receiver** before submitting an actual payment.
+
+Info endpoint takes both sender and receiver Libra IDs as arguments. Receiver VASP might decide to share more(or less) information based on sender ID, for example if the wallet supports some kind of contact list, they might decide to share more information if the sender is in the contact list of the receiver.
+
+**Request:**
+```json
+{
+   "sender_id": "<Sender Libra ID>",
```

Comment:
I wonder if we should allow the sender to optionally send additional info such as the user object. It removes the request the other direction to fetch that info and maybe a receiver is more likely to show such information like you mention if they are in their contact list.  So if the sender says they have phone number 123-4567 and that's in the user's contact book, maybe they are more likely to share info?  Not sure if this is a good idea or not, just something to think about